python/cudf/cudf/core/dataframe.py

```diff
@@ -263,7 +263,9 @@ def _init_from_dict_like(self, data, index=None, columns=None):
                 if is_scalar(data[col_name]):
                     num_rows = num_rows or 1
                 else:
-                    data[col_name] = column.as_column(data[col_name])
+                    data[col_name] = column.as_column(
+                        data[col_name], nan_as_null=True
```

Comment:
Does this not default to `nan_as_null=True`?

Reply:
Default param is `nan_as_null=None`,
We want to hit this case basically: https://github.com/rapidsai/cudf/blob/branch-0.14/python/cudf/cudf/core/column/column.py#L1348